src/renderer/components/dialogs/ProjectInvite.js

```diff
@@ -0,0 +1,192 @@
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import { Button, makeStyles, Typography } from '@material-ui/core'
+import Radio from '@material-ui/core/Radio'
+import RadioGroup from '@material-ui/core/RadioGroup'
+import FormControlLabel from '@material-ui/core/FormControlLabel'
+import FormControl from '@material-ui/core/FormControl'
+import { CopyToClipboard } from '../CopyToClipboard'
+import QRCode from 'react-qr-code'
+import Divider from '@material-ui/core/Divider'
+import { SettingsContext } from '../Settings/SettingsContext'
+
+const m = defineMessages({
+  // Title of modal asking user to confirm that they are leaving practice mode
+  titleLeavePractice: 'Leave Practice Mode',
+  // Title of join project modal
+  titleJoinProject: 'Join Project',
+  // Title of modal confirming that the user was successfully invited to project
+  titleProjectJoined: "You've Been Invited",
+  // Asks user to confirm if the would like to bring their observations from practice mode
+  bringObservations: 'Bring observations from Practice Mode to the project',
+  // Confirmation that user would like to keep observations when joining a new project
+  keepObservations: 'Yes, keep my observations',
+  // Confirmation that user would like to discard observations when joining a new project
+  deleteObservations: 'No, delete my observations',
+  // button text to cancel action
+  cancel: 'Cancel',
+  // instructions for user to show qr code to the project admin
+  showQr: 'Show this QR code to a Project Coordinator',
+  // alternative instructions for user send a join request instead of using the QR code
+  sendJoinRequest: 'Or Send Join Request',
+  // instructions for user to send a join request
+  copyUrl: 'Copy the url and send it to the Project Coordinator',
+  // button text to close dialog,
+  close: 'Close'
+})
+
+/**
+ * @typedef ProjectInviteDialogProps
+ * @prop {boolean} isOpen
+ * @prop {function(boolean?):void} toggleOpenClose
+ */
+
+/** @param {ProjectInviteDialogProps} props */
+export const ProjectInviteDialog = ({ isOpen, toggleOpenClose }) => {
+  return (
+    <Dialog open={isOpen}>
+      <CustomDialogContent closeDialog={() => toggleOpenClose(false)} />
+    </Dialog>
+  )
+}
+
+/**
+ * @typedef CustomDialogContentProps
+ * @prop {function():void} closeDialog
+ */
+
+/** @param {CustomDialogContentProps} props */
+const CustomDialogContent = ({ closeDialog }) => {
+  const { formatMessage: t } = useIntl()
+
+  const { practiceModeOn } = React.useContext(SettingsContext)
+
+  /** @type {[boolean|null, function]} */
+  const [migrateObservations, setMigrateObservations] = React.useState(null)
+
+  const [migrationPlanConfirmed, setMigrationPlanConfirmed] = React.useState(
+    false
+  )
+
+  const classes = useStyles()
+
+  if (practiceModeOn && !migrationPlanConfirmed) {
+    return (
+      <React.Fragment>
```

Comment:
probably has minimal effect but just be somewhat weary of using fragments to wrap entire components because it subjects it to being less predictably styled by the parent. not sure if an issue in this case though

Reply:
These component are specifically going to be wrapped in a dialog (they are not exported components, so used very specifically), so i think in this case it is not an issue!